### DISCUSSION TITLE
may be of interest: popsicle project

### DISCUSSION THREAD

**Discussion by shakfu** [DBraun/DawDreamer#110] [General]
Recently fell upon the [popsicle project](https://github.com/kunitoki/popsicle) which provides a python interface to the JUCE framework via [cppyy](https://github.com/wlav/cppyy) instead of [pybind11](https://github.com/pybind/pybind11).

It describes itself as follows

> Popsicle is a project that aims to give JUCE (https://juce.com/) a broader audience by allowing it to be used from python. Thanks to cppyy (http://cppyy.readthedocs.io/en/latest/) it exposes the JUCE framework api in a pythonic way, and the way it enables to write apps in python is very much similar to the way of writing them in C++ but without the overweight of managing project build, configurations and IDE solutions.

Not sure whether this could be of use to dawdreamer users. Perhaps for ad-hoc audio visualization, etc. ?

**Comment by shakfu** [comment]
Attached is a somewhat crude demo of dawdreamer and popsicle working together.

You will need to `pip install popsicle` and of course `pip install dawdreamer`

[demo.py.zip](https://github.com/DBraun/DawDreamer/files/9033787/demo.py.zip)

**Reply by DBraun** [reply]
Cool! Thanks for sharing. Pressing Load twice causes it to crash. Other than that, I would be happy to accept a PR if you add this to the examples folder and also make a README next to it describing it a bit more.

**Reply by shakfu** [reply]
Thanks!  That was more of a throwaway proof-of-concept. I'll see if I can make something a little more worthwhile. It's still early days. (-:

**Comment by shakfu** [comment]
Here's another iteration to resolve the earlier segfaulting bug. Also added some command line options to demo the audio player with something more visually interesting than the audio file generated by the sine wav generator.

[demo.2.zip](https://github.com/DBraun/DawDreamer/files/9033913/demo.2.zip)

**Comment by shakfu** [comment]
@DBraun You may be interested in the [new release](https://github.com/kunitoki/popsicle/releases/tag/v0.9.6) from [popsicle](https://github.com/kunitoki/popsicle) which is now completely re-written and `pybind11`-based.

Does this make it easier for interop between Daydreamer and popsicle?
